receiver/snmpreceiver/scraper.go

```diff
@@ -161,7 +163,35 @@ func (s *snmpScraper) scalarDataToMetric(
 	// the metric config's attribute values.
 	dataPointAttributes := getScalarDataPointAttributes(configHelper, data.oid)
 
-	return addMetricDataPointToResource(data, metricHelper, configHelper, metricName, generalResourceKey, dataPointAttributes)
+	// Get resource attributes
+	resourceAttributes, err := getResourceAttributes(configHelper, data.oid, "0", map[string]indexedAttributeValues{}, scalarOIDScalarResourceAttributeValues)
+	if err != nil {
+		return fmt.Errorf(errMsgOIDResourceAttributeEmptyValue, metricName, err)
+	}
+
+	// Create a resource key using all of the relevant resource attribute names 
+	resourceAttributeNames := configHelper.getResourceAttributeNames(data.oid)
+
+	if len(resourceAttributeNames) > 0 {
+		resourceKey := getResourceKey(resourceAttributeNames, "0")
+
+		// Create a new resource if needed
+		resource := metricHelper.getResource(resourceKey)
```

Comment:
could have some refactoring to have the lines here down be outside of the if/else (they're pretty much the same)

Reply:
Yep, good idea